src/textual/_styles_cache.py

```diff
@@ -275,20 +295,42 @@ def post(segments: Iterable[Segment]) -> Iterable[Segment]:
         line: Iterable[Segment]
         # Draw top or bottom borders (A)
         if (border_top and y == 0) or (border_bottom and y == height - 1):
+            is_top = y == 0
             border_color = base_background + (
-                border_top_color if y == 0 else border_bottom_color
+                border_top_color if is_top else border_bottom_color
```

Comment:
There is a significant amount of work done here which may not be required if there is no label.

Can we short-circuit this for the most common cases where there is no border or no title?

Reply:
I don't follow!
`border_color` creates `border_color_as_style` which is needed by `get_box` regardless of whether there is a label or not.
(I just renamed a couple of things.)

So, I don't see what can be short-circuited here.

Reply:
By "here" I meant the code that follows, not necessarily just this line. If there was no label, could you avoid calling that `render_border_label` entirely (and possibly avoid calculating some of its parameters)?

Reply:
Commit e05e50b should do it.